docs/upgrade.md

```diff
@@ -0,0 +1,245 @@
+---
+title: 'Upgrade to Storybook 7'
+---
+
+Storybook 7 is our first major release in over 2 years. A lot has changed during that time, so this is a big one. This guide is meant to help you upgrade successfully!
+
+## Major breaking changes
+
+The frontend ecosystem moves quickly, so we’ve accumulated a lot of breaking changes in Storybook 7. The rest of this guide is intended to help you migrate successfully. However, there are a few major breaking changes that you should know about before you go further:
+
```

Comment:
@kylegach can you polish this section a bit more as the way it's written is basically saying the same thing as what's in the intro above.